src/scapy/layers/ssl_tls_crypto.py

```diff
@@ -36,21 +35,10 @@ def prf(master_secret, label, data):
 def x509_extract_pubkey_from_der(der_certificate):
     # Extract subjectPublicKeyInfo field from X.509 certificate (see RFC3280)
     cert = DerSequence()
-    cert.decode(der_certificate)                
-    
+    cert.decode(der_certificate)
     tbsCertificate = DerSequence()
-    tbsCertificate.decode(cert[0])       # first DER SEQUENCE

```

Comment:
reverts 048c353eb23cbf875327c3da973a340cba3c5362
relying on tbscertificate[6] is error prone; the fix from 048c353eb23cbf875327c3da973a340cba3c5362 does a quick search for the OID 'rsaEncryption'